tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/EagerSession.java

```diff
@@ -282,6 +289,21 @@ public OperationBuilder opBuilder(String type, String name) {
     return new EagerOperationBuilder(this, type, name);
   }
 
+  @Override
+  public void attachFunction(ConcreteFunction function) {
+    checkSession();
+    try (PointerScope scope = new PointerScope()) {
+      attachNativeFunction(function.nativeHandle());
+      function.getDependencies().forEach(this::attachNativeFunction);
+    }
+  }
+
+  private void attachNativeFunction(TF_Function fn) {
```

Comment:
maybe move the entire code of this mode in the unique place it is called (i.e. in `attachFunction`) so we are sure it is never being called outside a `PointerScope`?